core/domain/exp_domain.py

```diff
@@ -3546,9 +3546,74 @@ def _convert_states_v22_dict_to_v23_dict(cls, states_dict):
                 state_dict, html_cleaner.add_caption_attr_to_image)
         return states_dict
 
+    @classmethod
+    def _convert_states_v23_dict_to_v24_dict(cls, states_dict, exp_id):
+        """Converts from version 23 to 24. Version 24 adds the dimensions of
+        images in the oppia-noninteractive-image tags.
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for key, state_dict in states_dict.iteritems():
+            state_dict['content']['html'] = (
+                html_cleaner.add_dimensions_to_image(
+                    exp_id, (state_dict['content']['html'])))
+
+            if state_dict['interaction']['default_outcome']:
+                interaction_feedback_html = state_dict[
+                    'interaction']['default_outcome']['feedback']['html']
+                state_dict['interaction']['default_outcome']['feedback'][
+                    'html'] = html_cleaner.add_dimensions_to_image(
+                        exp_id, (interaction_feedback_html))
+
+            for answer_group_index, answer_group in enumerate(
+                    state_dict['interaction']['answer_groups']):
+                answer_group_html = answer_group['outcome']['feedback']['html']
+                state_dict['interaction']['answer_groups'][
+                    answer_group_index]['outcome']['feedback']['html'] = (
+                        html_cleaner.add_dimensions_to_image(
+                            exp_id, (answer_group_html)))
+                if state_dict['interaction']['id'] == 'ItemSelectionInput':
+                    for rule_spec_index, rule_spec in enumerate(
+                            answer_group['rule_specs']):
+                        for x_index, x in enumerate(rule_spec['inputs']['x']):
+                            state_dict['interaction']['answer_groups'][
+                                answer_group_index]['rule_specs'][
+                                    rule_spec_index]['inputs']['x'][x_index] = html_cleaner.add_dimensions_to_image(exp_id, (x)) # pylint: disable=line-too-long
```

Comment:
Nope, this is an abuse of pylint-disable. Use intermediate variables.

Reply:
Done!